types.bal

```diff
@@ -0,0 +1,281 @@
+    string triggerType = "";
+    *Common;
+|};
+
+type JsonMap map<json>;
+
+# Represents the record type which contain necessary elements for a query.
+# 
+# + query - The SQL query represented as string.
+# + parameters - Parameters of the query if exists.
+public type Query record {|
+    string query = "";
+    QueryParameter[]? parameters = [];
+|};
+
+# Represents the paramaters related to query.
+# 
+# + name - Name of the parameter.
+# + value - Value of the parameter.
+public type QueryParameter record {|
+    string name = "";
+    string|int|boolean value = "";
+|};
+
+// ---------------------------------------------Managment Plane---------------------------------------------------------
+
+# Reprsent the record type with necessary paramaters to create partition key range.
+# 
+# + id - ID for the partition key range.
+# + minInclusive - Minimum partition key hash value for the partition key range. 
+# + maxExclusive - Maximum partition key hash value for the partition key range. 
+# + status - 
+public type PartitionKeyRange record {|
+    string id = "";
+    string minInclusive = "";
+    string maxExclusive = "";
+    string status = "";
+    *Common;
+|};
+
+# Represent the record type with necessary parameters to represent a user.
+# 
+# + id - User generated unique ID for the user. 
+# + permissions - Addressable path of the permissions resource.
+public type User record {|
+    string id = "";
+    string permissions?;
+    *Common;
+|};
+
+# Represent the record type with necessary parameters to represent a permission.
+# 
+# + id - User generated unique ID for the permission.
+# + permissionMode - Access mode for the resource, "All" or "Read".
+# + resourcePath - Full addressable path of the resource associated with the permission.
+# + validityPeriod - Optional. Validity period of the resource token.
+# + token - System generated resource token for the particular resource and user.
+public type Permission record {|
+    string id = "";
+    string permissionMode = "";
+    string resourcePath = "";
+    int validityPeriod?;
+    string token?;
+    *Common;
+|};
+
+# Represent the record type with necessary parameters to represent an offer.
+# 
+# + id - User generated unique ID for the offer.
+# + offerVersion - Offer version, This value can be V1 for pre-defined throughput levels and V2 for user-defined throughput levels.
+# + offerType - Optional. Performance level for V1 offer version, allows S1,S2 and S3.
+# + content - Information about the offer.
+# + resourceResourceId - The resource id(_rid) of the collection.
+# + resourceSelfLink - The self-link of the collection.
+public type Offer record {|
+    string id = "";
+    string offerVersion = "";
+    string? offerType?;
+    json content = {};
+    string resourceResourceId = "";
+    string resourceSelfLink = "";
+    *Common;
+|};
+
+
+class StreamGenerator {
+    public isolated function next() returns record {}|error? {
+        
+    }
+}
```

Comment:
Add a new line at the end of the file

Reply:
New lines added in the code is automatically removed by GitHub. Don't know exact reason for it.